logging/level.go

```diff
@@ -4,13 +4,38 @@ import (
 	"encoding/json"
 	"fmt"
 	"strings"
+	"sync/atomic"
 
 	"go.uber.org/zap/zapcore"
 )
 
 // Level is an enum of log levels. Its value can be `DEBUG`, `INFO`, `WARN` or `ERROR`.
 type Level int
 
+// AtomicLevel is a level that can be concurrently accessed.
+type AtomicLevel struct {
+	val *atomic.Int32
+}
+
+// NewAtomicLevelAt creates a new AtomicLevel at the input `initLevel`.
+func NewAtomicLevelAt(initLevel Level) AtomicLevel {
+	ret := AtomicLevel{
+		new(atomic.Int32),
+	}
+	ret.Set(initLevel)
+	return ret
+}
+
+// Set changes the level.
+func (level AtomicLevel) Set(newLevel Level) {
+	level.val.Store(int32(newLevel))
```

Comment:
[opt-nit] nil-guard for the atomic level pointer? I don't think `NewAtomicLevelAt`-created `AtomicLevel`s will have `nil` `val`s, but maybe worth a check to avoid
```
al := AtomicLevel{}
al.Set(DEBUG) // panic
```

Reply:
What would you have the code do if it saw a nil value?

Reply:
Ah 
```
func (level AtomicLevel) Set(newLevel Level) {
   if level.val == nil {
       level.val = new(atomic.Int32)
   }
   level.val.Store(newLevel)
}
```

Reply:
Because this is a shared piece of memory, that would be a data race with concurrent `Set`/`Get` calls. We'd need to introduce a mutex.

Reply:
Oops good point; ignore my request.